File: common/src/main/java/com/ssblur/scriptor/word/descriptor/AfterCastDescriptor.java
package com.ssblur.scriptor.word.descriptor;

import com.ssblur.scriptor.helpers.targetable.Targetable;

public interface AfterCastDescriptor {
  /**
   * The method to call after casting.
   * This is called after the initial cast, not after the subject is returned.
   * @param caster A Targetable representing this spell's caster
   */
  void afterCast(Targetable caster);
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/descriptor/AfterCastDescriptor.kt
package com.ssblur.scriptor.word.descriptor

import com.ssblur.scriptor.helpers.targetable.Targetable

interface AfterCastDescriptor {
    /**
     * The method to call after casting.
     * This is called after the initial cast, not after the subject is returned.
     * @param caster A Targetable representing this spell's caster
     */
    fun afterCast(caster: Targetable?)
}
